x/ecocredit/client/testsuite/tx.go

```diff
@@ -1426,6 +1428,81 @@ func (s *IntegrationTestSuite) TestTxBuyDirect() {
 	}
 }
 
+func (s *IntegrationTestSuite) TestUpdateProjectMetadata() {
+	val0 := s.network.Validators[0]
+	valAddrStr := val0.Address.String()
+	clientCtx := val0.ClientCtx
+	cmd := coreclient.TxUpdateProjectMetadataCmd()
+	_, projectId, _ := s.createClassProjectBatch(clientCtx, valAddrStr)
```

Comment:
With all these integration tests, I'm not sure why we are creating a new credit class, project, and batch for each test scenario. This is what we should be doing in a unit test, not necessarily an integration test. This also slows down our CI and tests. It's also not necessary to create a batch for this scenario.

Reply:
i could refactor to have some safety classes/projects that wont get trampled by other tests. the intention of the pattern was to avoid cases where other tests corrupt data needed by another test. 

> This is what we should be doing in a unit test, not necessarily an integration test

I don't think its necessarily detrimental thing to attempt to isolate individual integration test - its definitely a bit harder to achieve in a performant way with testing big systems like this though 🤷🏻 

> not necessary to create a batch for this scenario.

[chore: use new helper method, rename variables](https://github.com/regen-network/regen-ledger/pull/1105/commits/f15412e6d8561518a285494f1b1e12722a8f86fd), removes the batch creation part 👍🏻

Reply:
Removing batch creation helps. Not sure if we should spend too much time on this and attempt a larger refactor given we are planning to move away from client integration tests. Isolate and integration seem contradictory to me but that would be assuming integration tests are testing the integration of our features, not the integration with the sdk.